### PR TITLE
Propagate ARIA attributes to real-elements

### LINKED ISSUE #5489: [A11Y]:  <amp-img>, <amp-anim> should propagate `aria-*` to the inner `<img>`
Currently
`<amp-img aria-label="Some Image"  ..` does NOT read "Some Image" when using screen readers the same way `` to `` since the `` is the real component here.

**IMPORTANT**
I believe it is fine to leave the aria attributes on `<amp-img>` and just propagate them to ``, if we realize we need to actually remove the attributes from `<amp-img>` then we have to be careful to update places such as `<amp-image-lightbox>` that look for attributes such as `aria-describedby` on the customer element.

**To be on the safe side**, we can start by only propagating `aria-label`, `aria-labelledby` and `aria-describedby`

### PR DESCRIPTION
**Pull request by mkhatib**
Fixes #5489 and #5493

### CODE REVIEW

File: builtins/amp-img.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ import {BaseElement} from '../src/base-element';
 import {isLayoutSizeDefined} from '../src/layout';
 import {registerElement} from '../src/custom-element';
 import {srcsetFromElement} from '../src/srcset';
+import {dev} from '../src/log';
```
Comment (aghassemi) [inline builtins/amp-img.js]: nit: sort

File: builtins/amp-img.js [hunk 2]
```diff
@@ -70,7 +71,18 @@ export class AmpImg extends BaseElement {
     if (this.element.id) {
       this.img_.setAttribute('amp-img-id', this.element.id);
     }
-    this.propagateAttributes(['alt', 'referrerpolicy'], this.img_);
+
+    // Remove role=img otherwise this breaks screen-readers focus and
+    // only read "Graphic" when using only 'alt'.
+    if (this.element.getAttribute('role') == 'img') {
+      this.element.removeAttribute('role');
+      dev().error('Setting role=img on amp-img elements break screen readers ' +
```
Comment (aghassemi) [inline builtins/amp-img.js]: nit: s/break/breaks

File: examples/article.amp.html [hunk 3]
```diff
@@ -271,8 +271,8 @@
             src="img/hero@1x.jpg"
             srcset="img/hero@1x.jpg 1x, img/hero@2x.jpg 2x"
             layout="responsive" width="360" placeholder
-            alt="Curabitur convallis, urna quis pulvinar feugiat, purus diam posuere turpis, sit amet tincidunt purus justo et mi."
-            height="216" on="tap:headline-img-lightbox" role="img" tabindex="0">
+            alt="Two chairs on a lake shore with big pine trees"
+            height="216" on="tap:headline-img-lightbox" tabindex="0" role="img">
```
Comment (aghassemi) [inline examples/article.amp.html]: Let's remove `role="img"` and add `opens in a lightbox` to the end of the `alt`.

File: extensions/amp-anim/0.1/amp-anim.js [hunk 4]
```diff
@@ -16,9 +16,10 @@
 
 import {isLayoutSizeDefined} from '../../../src/layout';
 import {srcsetFromElement} from '../../../src/srcset';
+import {dev} from '../../../src/log';
```
Comment (aghassemi) [inline extensions/amp-anim/0.1/amp-anim.js]: nit: ditto sot

File: extensions/amp-anim/0.1/amp-anim.js [hunk 5]
```diff
@@ -41,9 +42,19 @@ class AmpAnim extends AMP.BaseElement {
...
-    this.propagateAttributes(['alt'], this.img_);
+    this.propagateAttributes(['alt', 'aria-label',
+      'aria-describedby', 'aria-labelledby'], this.img_);
     this.applyFillContent(this.img_, true);
 
+    // Remove role=img otherwise this breaks screen-readers focus and
+    // only read "Graphic" when using only 'alt'.
+    if (this.element.getAttribute('role') == 'img') {
+      this.element.removeAttribute('role');
+      dev().error('Setting role=img on amp-anim elements break screen ' +
```
Comment (aghassemi) [inline extensions/amp-anim/0.1/amp-anim.js]: ditto same comment on `amp-img`

File: builtins/amp-img.js (hunk 2 above)
Comment (jridgewell) [inline builtins/amp-img.js]: This is a user error, not a dev.

File: examples/article.amp.html (hunk 3 above)
Comment (mkhatib) [inline examples/article.amp.html]: Done

File: builtins/amp-img.js (hunk 2 above)
Comment (mkhatib) [inline builtins/amp-img.js]: Done

File: builtins/amp-img.js (hunk 1 above)
Comment (mkhatib) [inline builtins/amp-img.js]: Done

**Pull request merged**